docs/source/format/index.rst

```diff
@@ -22,17 +22,38 @@ Specifications
 
 .. toctree::
    :maxdepth: 2
+   :caption: Arrow Columnar Format
+   :hidden:
 
-   Versioning
+   FormatIntro
    Columnar
```

Comment:
@jorisvandenbossche haven't removed the IPC to a separate file as the opposite was done [here](https://github.com/apache/arrow/pull/5202/files#diff-6569a570a17ffd52bbe0b4f58cc6da2610ede6ac9c9d0771f46a01e80202946e) and the _IPC.rst_ is now an orphaned site.

Reply:
After talking to @jorisvandenbossche we decided to open an issue for moving the IPC spec content out of the Arrow Columnar format page: https://github.com/apache/arrow/issues/41671